libraries/SD/src/SD.h

```diff
@@ -84,35 +84,44 @@ class SDClass {
     }
 
     uint8_t type() {
-        return 0;//card.type();
+        sdfs::SDFSImpl* sd = static_cast<sdfs::SDFSImpl*>(SDFS.getImpl().get());
+        return sd->type();
     }
 
     uint8_t fatType() {
-        return 0;//volume.fatType();
+        sdfs::SDFSImpl* sd = static_cast<sdfs::SDFSImpl*>(SDFS.getImpl().get());
+        return sd->fatType();
     }
 
     size_t blocksPerCluster() {
-        return 0;//volume.blocksPerCluster();
+        sdfs::SDFSImpl* sd = static_cast<sdfs::SDFSImpl*>(SDFS.getImpl().get());
+        return sd->blocksPerCluster();
     }
 
     size_t totalClusters() {
-        return 0;//volume.clusterCount();
+        sdfs::SDFSImpl* sd = static_cast<sdfs::SDFSImpl*>(SDFS.getImpl().get());
+        return sd->totalClusters();
     }
 
     size_t blockSize() {
         return 512;
     }
 
     size_t totalBlocks() {
-        return 0;//(totalClusters() / blocksPerCluster());
+        return (totalClusters() / blocksPerCluster());
     }
 
     size_t clusterSize() {
-        return 0;//blocksPerCluster() * blockSize();
+        return blocksPerCluster() * blockSize();
     }
 
     size_t size() {
-        return 0;//(clusterSize() * totalClusters());
+        uint64_t sz = size64();
+	return (size_t)std::min( (uint64_t)SIZE_MAX, sz );
```

Comment:
This is misleading and is likely to be cause for future issues. Instead, I suggest returning 0 when overflowed (i. e. failed)

Reply:
and maybe print out a debug message.

Reply:
I'd argue the opposite.  Returning 0 is bound to cause trouble (0 means broken card/etc. with this fcn).  Returning 4GB when cards are >4GB are detected at least gives a size that is less likely to cause trouble.  Think of it as a saturating value as used in audio processing.

A warning printout seems good here, though, agreed.